prov/rxm/src/rxm_fabric.c

```diff
@@ -82,51 +82,64 @@ static struct fi_ops rxm_fabric_fi_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
-static int rxm_fabric_init_util_coll(struct rxm_fabric *fabric)
+static int rxm_fabric_init_coll(const char *name,
+				struct fi_info **coll_info,
+				struct fid_fabric **coll_fabric)
 {
-	struct fi_info *hints, *coll_info;
-	struct fid_fabric *coll_fabric;
+	struct fi_info *hints, *info;
+	struct fid_fabric *fabric;
 	int ret;
 
 	hints = fi_allocinfo();
 	if (!hints)
 		return -FI_ENOMEM;
 
-	hints->fabric_attr->prov_name = strdup(OFI_OFFLOAD_PREFIX "coll");
+	hints->fabric_attr->prov_name = strdup(name);
 	if (!hints->fabric_attr->prov_name) {
 		fi_freeinfo(hints);
 		return -FI_ENOMEM;
 	}
 
 	hints->mode = FI_PEER_TRANSFER;
 	ret = fi_getinfo(OFI_VERSION_LATEST, NULL, NULL, OFI_OFFLOAD_PROV_ONLY,
-			 hints, &coll_info);
+			 hints, &info);
 	fi_freeinfo(hints);
 
 	if (ret)
 		return ret;
 
-	ret = fi_fabric(coll_info->fabric_attr, &coll_fabric, NULL);
+	ret = fi_fabric(info->fabric_attr, &fabric, NULL);
 	if (ret)
 		goto err1;
 
-	fabric->util_coll_info = coll_info;
-	fabric->util_coll_fabric = coll_fabric;
-	return 0;
+	*coll_info = info;
+	*coll_fabric = fabric;
+	return FI_SUCCESS;
 
 err1:
-	fi_freeinfo(coll_info);
+	fi_freeinfo(info);
 	return ret;
 }
 
+static int rxm_fabric_init_util_coll(struct rxm_fabric *fabric)
+{
+	return rxm_fabric_init_coll(OFI_OFFLOAD_PREFIX "coll",
+				    &fabric->util_coll_info,
+				    &fabric->util_coll_fabric);
+}
+
 static int rxm_fabric_init_offload_coll(struct rxm_fabric *fabric)
 {
-	/*
-	 * TODO:
-	 * silimar to rxm_fabric_init_util_coll, except that the offload
-	 * provider is discovered by feature instead of name.
-	 */
-	return 0;
+	char *offload_coll_name;
```

Comment:
Initialize to NULL.  I don't think fi_param_get_str modifies the string if it's not defined.

Reply:
Done.